src/twisted/mail/imap4.py

```diff
@@ -127,27 +172,42 @@ def _setLast(self, value):
 
             self._last = value
             for i, (l, h) in enumerate(self.ranges):
-                if l is not None:
-                    break # There are no more Nones after this
-                l = value
+                if l is None:
+                    l = value
                 if h is None:
                     h = value
                 if l > h:
                     l, h = h, l
                 self.ranges[i] = (l, h)
-
             self.clean()
 
         def _getLast(self):
             return self._last
 
         doc = '''
-            "Highest" message number, referred to by "*".
-            Must be set before attempting to use the MessageSet.
-        '''
+              Replaces all occurrences of "*".  This should be the
+              largest number in use.  Must be set before attempting to
+              use the MessageSet as a container.
+
+              @raises: L{ValueError} if a largest value has already
+                  been set.
+              '''
         return _getLast, _setLast, None, doc
     last = property(*last())
 
+    def _swap(self, this, that, ifIs):
```

Comment:
This doesn't use `self` at all and should therefore be a top-level function, not a method.